src/main/java/jcifs/smb1/netbios/NameServiceClient.java

```diff
@@ -0,0 +1,444 @@
+                            ans.hostName.srcHashCode = 0; // just has to be different
+                                                          // from other methods
+                            return ans;
+                        }
+                        break;
+                    case RESOLVER_WINS:
+                    case RESOLVER_BCAST:
+                        if( resolveOrder[i] == RESOLVER_WINS &&
+                                name.name != NbtAddress.MASTER_BROWSER_NAME &&
+                                name.hexCode != 0x1d ) {
+                            request.addr = NbtAddress.getWINSAddress();
+                            request.isBroadcast = false;
+                        } else {
+                            request.addr = baddr;
+                            request.isBroadcast = true;
+                        }
+
+                        n = RETRY_COUNT;
+                        while( n-- > 0 ) {
+                            try {
+                                send( request, response, RETRY_TIMEOUT );
+                            } catch( IOException ioe ) {
+                                if( log.level > 1 )
+                                    ioe.printStackTrace( log );
+                                throw new UnknownHostException( name.name );
+                            }
+                            if( response.received && response.resultCode == 0 ) {
+
+/* Before we return, in anticipation of this address being cached we must
+ * augment the addresses name's hashCode to distinguish those resolved by
+ * Lmhosts, WINS, or BCAST. Otherwise a failed query from say WINS would
+ * get pulled out of the cache for a BCAST on the same name.
+ */
+                                response.addrEntry[0].hostName.srcHashCode =
+                                                        request.addr.hashCode();
+                                return response.addrEntry[0];
+                            } else if( resolveOrder[i] == RESOLVER_WINS ) {
+                                /* If WINS reports negative, no point in retry
+                                 */
+                                break;
+                            }
+                        }
+                        break;
+                }
+            } catch( IOException ioe ) {
+            }
+        }
+        throw new UnknownHostException( name.name );
+    }
+    NbtAddress[] getNodeStatus( NbtAddress addr ) throws UnknownHostException {
+        int n, srcHashCode;
+        NodeStatusRequest request;
+        NodeStatusResponse response;
+
+        response = new NodeStatusResponse( addr );
+        request = new NodeStatusRequest(
+                            new Name( NbtAddress.ANY_HOSTS_NAME, 0x00, null));
+        request.addr = addr.getInetAddress();
+
+        n = RETRY_COUNT;
+        while( n-- > 0 ) {
+            try {
+                send( request, response, RETRY_TIMEOUT );
+            } catch( IOException ioe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)